File: EIDSS v5/android/EIDSS/OptionsActivity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using eidssdroid.Model;
using eidssdroid.model.Database;

namespace EIDSS
{
    [Activity]
    public class OptionsActivity : Activity
    {
        protected override void OnCreate(Bundle bundle)
        {
            base.OnCreate(bundle);

            SetTitle(Resource.String.TitleSettings);
            SetContentView(Resource.Layout.OptionsLayout);
            Button buttonOk = FindViewById<Button>(Resource.Id.OkButton);
            Button buttonCancel = FindViewById<Button>(Resource.Id.CancelButton);
            Button buttonLoadRef = FindViewById<Button>(Resource.Id.LoadReferencesButton);
            Button buttonSetPwd = FindViewById<Button>(Resource.Id.SetPasswordButton);
            EditText url = FindViewById<EditText>(Resource.Id.ServerUrl);

            using (EidssDatabase db = new EidssDatabase(this))
            {
                url.Text = db.Options().strServerUrl;
            }

            buttonLoadRef.Click += delegate
            {
                StartActivity(typeof(LoadReferencesActivity));
            };

            buttonSetPwd.Click += delegate
            {
                StartActivity(typeof(ChangeLocalPasswordActivity));
            };

            buttonOk.Click += delegate
            {
                using (EidssDatabase db = new EidssDatabase(this))
                {
                    Options o = db.Options();
                    o.strServerUrl = url.Text;
                    db.OptionsUpdate(o);
                }
                this.Finish();
            };

            buttonCancel.Click += delegate
            {
                this.Finish();
            };
        }


    }
}

File: EIDSS v5/android.java/workspace/EIDSS/src/com/bv/eidss/OptionsActivity.java
package com.bv.eidss;

import android.os.Bundle;
import android.app.Activity;
import android.content.Intent;
import android.view.View;
import android.widget.EditText;

import com.bv.eidss.data.EidssDatabase;
import com.bv.eidss.model.Options;

public class OptionsActivity extends Activity {

	private Activity _this;
	public OptionsActivity() {
    	_this = this;
	}
	
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setTitle(R.string.TitleSettings);
        setContentView(R.layout.options_layout);
   	
        final EditText url = (EditText)findViewById(R.id.ServerUrl);
        
        EidssDatabase db = new EidssDatabase(this);
        url.setText(db.Options().getServerUrl());
        db.close();
        
    	findViewById(R.id.OkButton).setOnClickListener(new View.OnClickListener() {
			public void onClick(View v) {
		        EidssDatabase db = new EidssDatabase(_this);
                Options o = db.Options();
		        o.setServerUrl(url.getText().toString());
                db.OptionsUpdate(o);
                db.close();
				_this.finish();
			}
		});
    	findViewById(R.id.CancelButton).setOnClickListener(new View.OnClickListener() {
			public void onClick(View v) {
				_this.finish();
			}
		});
    	findViewById(R.id.LoadReferencesButton).setOnClickListener(new View.OnClickListener() {
			public void onClick(View v) {
				_this.startActivity((new Intent()).setClass(getApplicationContext(), LoadReferencesActivity.class));
			}
		});
    	findViewById(R.id.SetPasswordButton).setOnClickListener(new View.OnClickListener() {
			public void onClick(View v) {
				_this.startActivity((new Intent()).setClass(getApplicationContext(), ChangeLocalPasswordActivity.class));
			}
		});
    	
    }
	
}
